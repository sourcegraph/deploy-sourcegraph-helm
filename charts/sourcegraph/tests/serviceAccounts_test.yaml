suite: serviceAccount
release:
  name: sourcegraph
  namespace: sourcegraph
tests:
- it: should render service account when openTelemetry.gateway.serviceAccount.create is true
  set:
    openTelemetry:
      gateway:
        serviceAccount:
          create: true
  template: otel-collector/otel-collector.ServiceAccount.yaml
  asserts:
  - isKind:
      of: ServiceAccount
  - equal:
      path: metadata.name
      value: otel-collector
  - matchSnapshot: {}

- it: should should reference service account when openTelemetry.gateway.serviceAccount.create is true
  set:
    openTelemetry:
      gateway:
        serviceAccount:
          create: true
  template: otel-collector/otel-collector.Deployment.yaml
  asserts:
  - equal:
      path: spec.template.spec.serviceAccountName
      value: otel-collector

- it: should render service account when openTelemetry.agent.serviceAccount.create is true
  set:
    openTelemetry:
      agent:
        serviceAccount:
          create: true
  template: otel-collector/otel-agent.ServiceAccount.yaml
  asserts:
  - isKind:
      of: ServiceAccount
  - equal:
      path: metadata.name
      value: otel-agent
  - matchSnapshot: {}

- it: should should reference service account when openTelemetry.agent.serviceAccount.create is true
  set:
    openTelemetry:
      agent:
        serviceAccount:
          create: true
  template: otel-collector/otel-agent.DaemonSet.yaml
  asserts:
  - equal:
      path: spec.template.spec.serviceAccountName
      value: otel-agent

- it: should render service account when frontend.serviceAccount.create=true
  set:
    frontend:
      serviceAccount:
        create: true
  template: frontend/sourcegraph-frontend.ServiceAccount.yaml
  asserts:
  - isKind:
      of: ServiceAccount
  - equal:
      path: metadata.name
      value: sourcegraph-frontend
