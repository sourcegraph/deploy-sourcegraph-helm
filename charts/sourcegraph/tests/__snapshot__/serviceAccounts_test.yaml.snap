should render service account when frontend.serviceAccount.create=true:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/component: frontend
        category: rbac
        deploy: sourcegraph
      name: sourcegraph-frontend
should render service account when openTelemetry.agent.serviceAccount.create is true:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/component: otel-collector
        category: rbac
        deploy: sourcegraph
      name: otel-agent
should render service account when openTelemetry.gateway.serviceAccount.create is true:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        app.kubernetes.io/component: otel-collector
        category: rbac
        deploy: sourcegraph
      name: otel-collector
should should reference service account when openTelemetry.agent.serviceAccount.create is true:
  1: |
    apiVersion: apps/v1
    kind: DaemonSet
    metadata:
      annotations:
        description: Forwards telemetry data to the OpenTelemetry Collector Deployment.
      labels:
        app.kubernetes.io/component: otel-collector
        app.kubernetes.io/instance: sourcegraph
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: sourcegraph
        app.kubernetes.io/version: 4.4.2
        deploy: sourcegraph
        helm.sh/chart: sourcegraph-4.4.2
      name: otel-agent
    spec:
      minReadySeconds: 5
      selector:
        matchLabels:
          app: otel-agent
          app.kubernetes.io/instance: sourcegraph
          app.kubernetes.io/name: sourcegraph
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: otel-agent
          labels:
            app: otel-agent
            app.kubernetes.io/instance: sourcegraph
            app.kubernetes.io/name: sourcegraph
            deploy: sourcegraph
        spec:
          affinity: null
          containers:
          - command:
            - /bin/otelcol-sourcegraph
            - --config=/etc/otel-agent/config.yaml
            env: null
            image: index.docker.io/sourcegraph/opentelemetry-collector:4.4.2@sha256:f0723c96c973258ad3123ddc479261bb8f5827bbac1d091b6a683fde55334413
            imagePullPolicy: IfNotPresent
            livenessProbe:
              httpGet:
                path: /
                port: 13133
            name: otel-agent
            ports:
            - containerPort: 55679
              hostPort: 55679
              name: zpages
            - containerPort: 4317
              hostPort: 4317
              name: otlp-grpc
            - containerPort: 4318
              hostPort: 4318
              name: otlp-http
            readinessProbe:
              httpGet:
                path: /
                port: 13133
            resources:
              limits:
                cpu: 500m
                memory: 500Mi
              requests:
                cpu: 100m
                memory: 100Mi
            terminationMessagePolicy: FallbackToLogsOnError
            volumeMounts:
            - mountPath: /etc/otel-agent
              name: config
          nodeSelector: null
          serviceAccountName: otel-agent
          terminationGracePeriodSeconds: 120
          tolerations: null
          volumes:
          - configMap:
              items:
              - key: config.yaml
                path: config.yaml
              name: otel-agent
            name: config
should should reference service account when openTelemetry.gateway.serviceAccount.create is true:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      annotations:
        description: Receives, processes, and exports telemetry data.
      labels:
        app.kubernetes.io/component: otel-collector
        app.kubernetes.io/instance: sourcegraph
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: sourcegraph
        app.kubernetes.io/version: 4.4.2
        deploy: sourcegraph
        helm.sh/chart: sourcegraph-4.4.2
      name: otel-collector
    spec:
      minReadySeconds: 5
      progressDeadlineSeconds: 120
      replicas: 1
      selector:
        matchLabels:
          app: otel-collector
          app.kubernetes.io/instance: sourcegraph
          app.kubernetes.io/name: sourcegraph
      template:
        metadata:
          annotations:
            kubectl.kubernetes.io/default-container: otel-collector
          labels:
            app: otel-collector
            app.kubernetes.io/instance: sourcegraph
            app.kubernetes.io/name: sourcegraph
            deploy: sourcegraph
        spec:
          affinity: null
          containers:
          - command:
            - /bin/otelcol-sourcegraph
            - --config=/etc/otel-collector/configs/logging.yaml
            env: null
            image: index.docker.io/sourcegraph/opentelemetry-collector:4.4.2@sha256:f0723c96c973258ad3123ddc479261bb8f5827bbac1d091b6a683fde55334413
            imagePullPolicy: IfNotPresent
            livenessProbe:
              httpGet:
                path: /
                port: 13133
            name: otel-collector
            ports:
            - containerPort: 55679
              name: zpages
            - containerPort: 4317
              name: otlp-grpc
            - containerPort: 4318
              name: otlp-http
            - containerPort: 8888
              name: metrics
            readinessProbe:
              httpGet:
                path: /
                port: 13133
            resources:
              limits:
                cpu: "3"
                memory: 3Gi
              requests:
                cpu: "1"
                memory: 1Gi
            terminationMessagePolicy: FallbackToLogsOnError
            volumeMounts: null
          nodeSelector: null
          serviceAccountName: otel-collector
          terminationGracePeriodSeconds: 120
          tolerations: null
          volumes: null
